lib/api_repo.js

```diff
@@ -806,57 +831,85 @@ function processPaths(done, process) {
  * Finds the paths to the proto files with the given api name and version.
  *
  * If callback is set, it calls back on each of them.
- * @param {string} name the api name.
+ * @param {string} the api name
  * @param {string} version the api version
- * @param {function(?Error)} done the cb called with all the protos or an error
- * @param {function(?Error, string[])} onProto the callback called on each
- *   proto
+ * @param {function(string[], ?Error)} done the cb called with all the protos
+     or an error
+ * @param {function(string, string, function(?Error))} onProto the callback
+     called on each proto/protoDir pair
  */
 ApiRepo.prototype._findProtos = function _findProtos(name, version, done,
                                                      onProto) {
   // Determine the top-level proto dir
-  var topDir = this.repoDir;
-  var parts = [topDir];
+  var protoDirs = this.repoDirs;
+
   if (this.isGoogleApi) {
-    parts.push('google');
-  }
-  parts.push(name, version);
-  var protoDir = path.join.apply(path, parts);
-
-  /* Use glob to scan for protos in a protoDir */
-  function scanForProtos(next) {
-    var stripRoot = processPaths(next, function(paths) {
-      return _.map(paths, function(x) {
-        return x.replace(topDir + '/', '');
-      });
+    protoDirs = protoDirs.map(function (parts) {
+      return path.join(parts, 'google');
     });
+  }
 
-    glob.glob("*.proto", {
-      cwd: protoDir,
-      realpath: true,
-      nodir: true
-    }, stripRoot);
+  if (this.zipUrl) {
+    protoDirs = protoDirs.map(function (parts) {
+      return path.join(parts, name, version);
+    });
```

Comment:
If zipUrl is not specified, then the API name and version are already encoded into the repoDirs via the `-r` flag. However, if zipUrl is specified, then we only have the root directory that the URL points to. Maybe it would make more sense to take care of this in `_checkRepo`, but I don't currently...